execute_llvm/src/lib.rs

```diff
@@ -105,27 +105,39 @@ fn run_function<T>(
     parse_result(&ctx, llvm_result)
 }
 
-#[pyfunction]
-fn run_int_function(hugr_json: &str, fn_name: &str) -> PyResult<i64> {
-    run_function::<i64>(hugr_json, fn_name, |_, llvm_val| {
-        // GenericVal is 64 bits wide
-        let int_with_sign = llvm_val.as_int(true);
-        let signed_int = int_with_sign as i64;
-        Ok(signed_int)
-    })
-}
+#[pymodule]
+mod execute_llvm {
+    use inkwell::context::Context;
+    use pyo3::pyfunction;
 
-#[pyfunction]
-fn run_float_function(hugr_json: &str, fn_name: &str) -> PyResult<f64> {
-    run_function::<f64>(hugr_json, fn_name, |ctx, llvm_val| {
-        Ok(llvm_val.as_float(&ctx.f64_type()))
-    })
-}
+    use super::*;
 
-#[pymodule]
-fn execute_llvm(_py: Python, m: &PyModule) -> PyResult<()> {
-    m.add_function(wrap_pyfunction!(compile_module_to_string, m)?)?;
-    m.add_function(wrap_pyfunction!(run_int_function, m)?)?;
-    m.add_function(wrap_pyfunction!(run_float_function, m)?)?;
-    Ok(())
+    #[pyfunction]
+    fn compile_module_to_string(hugr_json: &str) -> PyResult<String> {
+        let mut hugr = parse_hugr(hugr_json)?;
+        let ctx = Context::create();
+
+        hugr = guppy_pass(hugr);
+        let module = compile_module(&hugr, &ctx, Default::default())?;
+
+        Ok(module.print_to_string().to_str().unwrap().to_string())
+    }
+
+    #[pyfunction]
+    fn run_int_function(hugr_json: &str, fn_name: &str) -> PyResult<i64> {
+        run_function::<i64>(hugr_json, fn_name, |_, llvm_val| {
+            // GenericVal is 64 bits wide
+            let int_with_sign = llvm_val.as_int(true);
+            let signed_int = int_with_sign as i64;
+            panic!("test 6");
```

Comment:
!

Reply:
-.-
Should this have triggered a CI fail?